src/app/core/permissions/entity-operation.directive.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  Directive,
+  ElementRef,
+  Input,
+  OnChanges,
+  Renderer2,
+} from "@angular/core";
+import {
+  EntityPermissionsService,
+  OperationType,
+} from "./entity-permissions.service";
+import { Entity } from "../entity/entity";
+
+@Directive({
+  selector: "[appEntityOperation]",
+})
+export class EntityOperationDirective implements OnChanges {
+  @Input("appEntityOperation") arguments: {
+    operation: OperationType;
+    entity: typeof Entity;
+  };
+  constructor(
+    private el: ElementRef,
+    private entityPermissionService: EntityPermissionsService,
+    private renderer: Renderer2
+  ) {}
+
+  ngOnChanges() {
+    if (this.arguments?.operation && this.arguments?.entity) {
+      const permitted = this.entityPermissionService.userIsPermitted(
+        this.arguments.entity,
+        this.arguments.operation
+      );
+      setTimeout(() => {
+        // The timeout is required because the mat-button directive otherwise sets disabled back to false in the
+        // Form component. Other components somehow do not have this problem.
+        if (!permitted) {
+          this.renderer.setAttribute(this.el.nativeElement, "disabled", "true");
```

Comment:
do we really have to go through the native element? Is there no way to set the disabled input on the component here programmatically?

Reply:
The use of the `Renderer2` by Angular should prevent this problem. Another approach would be to get rid of the directive and instead just set the `[disabled]` directive depending on the `userIsPermitted` value.